fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Memo.java

```diff
@@ -305,12 +304,14 @@ public void removeUnreachableGroup() {
         LinkedList<Integer> touch = new LinkedList<>();
         touch.add(rootGroup.getId());
         deepSearchGroup(rootGroup, touch);
+        groups.removeIf(g -> !touch.contains(g.getId()));
+        groupExpressions.clear();
 
-        List<Group> groupsCopy = new ArrayList<>(groups);
-        for (Group group : groupsCopy) {
-            if (!touch.contains(group.getId())) {
-                removeOneGroup(group);
-            }
+        // only used in logical rewrite phase, logical expression must only one in a group,
```

Comment:
I think if this function can only be used in logical rewrite phase, we should add  some comment for this function to avoid wrong use.